File: app/javascript/channels/home_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("HomeChannel", {
  connected() {
    console.log("welcome user connected");
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/home_channel.rb
class HomeChannel < ApplicationCable::Channel
  def subscribed
    stream_from "home_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    ActionCable.server.broadcast('home_channel', data)
  end

  def recieve(data)
    ActionCable.server.broadcast('home_channel', data)
  end
end